second ludo review of group 4 by group 2 and 7

skjohansen opened the pull request.

aohzork committed:
Driver @aohzork

PontusGillenang committed:
PairProgramming @aohzork @samirehsani19 @pontusgillenang

aohzork committed:
Driver@aohzork

samirehsani19 committed:
Driver @samirehsani19

installed Extention for json and .net core

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Driver @pontusgillenang

Begun testing

aohzork committed:
Driver@aohzork

samirehsani19 committed:
Driver @samirehsani19

Added json  file

PontusGillenang committed:
Driver @pontusgillenang

samirehsani19 committed:
driver @samirehsani19

Added connectionstring

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Solved error

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Driver @pontusgillenang

samirehsani19 committed:
Driver @samirehsani19

Created Properties

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
Driver @samirehsani19

Created DbSet for Piece

aohzork committed:
Driver @aohzork

aohzork committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
Driver @samirehsani19

Code Error solved

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Created Tables and solved problem with database

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

aohzork committed:
Driver@aohzork

muratkugic commented on Documentation/backlog_bugs.md:
```diff
@@ -0,0 +1,9 @@
+# Backlog bugs
+
+Här skrivs buggar och ev. saknad funktionalitet ner. När de är klara markeras de med "OK" innan början av texten. För att skapa en backlog av buggar att åtgärda
+
+======================================================================
+
+Huvudmeny- Välja färg, upptagen färg skall försvinna
+
+GameBoard - int DecidePlayerStart(int pAmount): saknas kod för att kontrollera att flera spelare slår samma högsta siffra på tärningen och måste slå om tills någon vinner.
```
Nice med backlog, enkelt att hålla reda på vad som behöver göras. Kanske prioritet på de bitar som är kvar? 👍

muratkugic commented on Documentation/dagbok_Micael_Wollter_aohzork.md:
```diff
@@ -12,4 +12,24 @@ Har tillsammans med gruppen Samir och Pontus skapat lowlevel flowchart för data
 
-Har skrivit klart samtliga userstories. Lagt upp egen förkortad version av spelregler.
\ No newline at end of file
+[natt]Har skrivit klart samtliga userstories. Lagt upp egen förkortad version av spelregler.
+
+[dag] Gemensam genomgång av att göra klart CRC-Cards och databasdesign. Delat upp i ansvarsområden och påbörjat kodning. Gjort meny för spelet. Alla metoder inte klara då databas behöver vara klart först.
+
+2020-04-02
+
+Gjort Fluent API för GameSession samt påbörjat GameBoard, GamePlayer och GamePiece klasser.
+
+2020-04-03 to 2020-04-04
+
+Fortsatt justering av GameSession samt GameBoard, GamePlayer & GamePice hur de skall hänga ihop med varandra och hämta data. Även påbörjat arbetet av koordinatsystemet för GamePiece samt GameBoard.
+
+Skapat diceklass,  lagt till metoder i GameBoard. Gjort Unit Test på Gameboard
+
+2020-04-05
+
+Jobbat med en turodningsmetod som bestämmer vilken ordning spelarna skall spela (medsols), beroende på vilken färg som börjar slå. Gjort unit-test, samt påbörjat gameplay
+
+2020-04-06
+
+Refakturerat gameplay. Gjort metoder kring regler då spelare slår 1-5. Kvarstår 6
```
👍

muratkugic commented on Documentation/process- och projektdagbok.md:
```diff
@@ -17,4 +17,15 @@ Målet är i slutändan att de skall resultera i CRC-cards innehållandes tillr
 **<u>I korthet processen:</u>**
-*Flowchart (Highlevel) -> Userstories -> Flowchart (Lowlevel) -> CRC-Cards -> Koda -> Iterera dokumentation -> koda os*v.
\ No newline at end of file
+*Flowchart (Highlevel) -> Userstories -> Flowchart (Lowlevel) -> CRC-Cards -> Koda -> Iterera dokumentation -> koda os*v.
+
+## Genomförandefasen
+
+** Nedan löper kontinuerliga uppdateringar för projektets status, samlat över en eller flera dagar. Datum indikerar då senaste uppdatering skett för att på så vis får spårbarhet i projektet.
+
+**[Update 1 - 2020-04-01]**
+
+Flowcaharts Lolevel är klara såväl som CRC-cards (första version) för samtliga delar såväl som databasdesign. Databasdesignen gjordes i MSSMS och utifrån den blir det lättare att bygga databasen mha Code First.
+
+Vi har initialt delat upp arbetet i olika ansvarsområden och påbörjat kodning.
+
```
Tydligt som Snören! :)  Kort och koncist - mycket //fint! 💯

muratkugic commented on Source/LudoGame/Menu.cs:
```diff
@@ -0,0 +1,149 @@
+            }
+        }
+    }
+}
```
Kommenterar hela filen: Snyggt med små metoder! Underlättar felsökning/optimering då det är små kodstycken att ta in! 💯

muratkugic commented on Source/LudoGameEngine/GameBoard.cs:
```diff
@@ -0,0 +1,292 @@
+
+            int convertToGlobalPosIndex = (GamePlayers[playerIndex].GlobalStartPos) + (GamePlayers[playerIndex].Pieces[pieceIndex].CurrentPos--);
+                       
+            return convertToGlobalPosIndex;
+
+            //CheckCollission
+            //Draw Graphics
+        }
+
+        private void RollSix(int stepsToMove, int playerIndex)
+        {
+
+        }
+
+        private int RollRegular(int stepsToMove, int playerIndex)
+        {
+            //dialoge here
+            //menu here ersätt med foreachloop
+
+            var piecesToMove = GamePlayers[playerIndex].Pieces.Where(s => s.CurrentPos != 0);
+            foreach(var p in piecesToMove)
+            {
+                Console.WriteLine("Avaliable pieces to move:");                
+                Console.Write($"ID:{p.PieceID} ");
+            }
+            Console.WriteLine("\n\rChoose a piece to move: ");
+            int id = int.Parse(Console.ReadLine());
+
+            var piece = GamePlayers[playerIndex].Pieces.Where(s => s.PieceID == id).FirstOrDefault();
+            int pieceIndex = GamePlayers[playerIndex].Pieces.IndexOf(piece);
+
+            int currentPos = GamePlayers[playerIndex].Pieces[pieceIndex].CurrentPos;
+            GamePlayers[playerIndex].Pieces[pieceIndex].LocalCoordinatePositions[currentPos] = false;
+            GamePlayers[playerIndex].Pieces[pieceIndex].CurrentPos += MovePiece(stepsToMove);
+            GamePlayers[playerIndex].Pieces[pieceIndex].LocalCoordinatePositions[currentPos] = true;
+
+            int convertToGlobalPosIndex = (GamePlayers[playerIndex].GlobalStartPos) + (GamePlayers[playerIndex].Pieces[pieceIndex].CurrentPos--);
+
+            return convertToGlobalPosIndex;
+        }
```
Känns som kodupprepning med dessa två metoder då de är rätt snarlika. Kanske kunde hantera detta via en mer genrell metod? : )

aohzork replied:
Japp :) Du har rätt! Tanken är att göra det. Skall bara ha alla metoder(dvs när man slår 6) på plats först, så jag ser hur kodeflödet ser ut, så skall jag göra en mer abstrakt metod att använda.

muratkugic commented on Source/LudoGameEngine/GameSession.cs:
```diff
@@ -0,0 +1,112 @@
+            return SessionPlayerData;
+        }
+    }
+}
```
Kommentar på hela filen: något att tänka på är att bryta loss alla Console.Write/Read och specifika kodstycken som har med Konsollen att göra (Gui element) till att vara i Ludo Game projektet. För nu har ni kopplat erat bibliotek till konsollen så det blir svårt att använda detta bibliotek till exempelvis WPF/ASP.NET CORE etc.. : )

muratkugic commented on Source/LudoGameEngine/Models/Player.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public class Player
+    {
+        [Key]
+        public int PlayerID { get; set; }
+        [Required(ErrorMessage ="Required")]
+        [StringLength (25, MinimumLength =3, ErrorMessage ="Invalid name length")]
+        public string Name { get; set; }
+        [Required(ErrorMessage = "Required")]
+        [StringLength(25, MinimumLength = 3, ErrorMessage = "Invalid color length")]
+        public string Color { get; set; }
+        public int WinCount { get; set; }
+        [Required(ErrorMessage ="Session ID is Required")]
+        public Session SessionId { get; set; }
+    }
+}
```
nice med annotations för databasen! tydligt

muratkugic commented on Source/XFunctionalTest/UnitTest1.cs:
```diff
@@ -1,9 +1,9 @@
+    public class UnitTest1
     {
         [Fact]
         public void Test1()
```
Nice med tester, såg bra ut det Stephan kollade på lektionen tidigare idag.

aohzork replied:
Iaf när han visade hur man kunde lösa testet :) Lärde oss en hel del

skjohansen replied:
antar at han är mig ;) i Github brukar man att göra en "mention" skriv "@" och välj person eller trykka på "@" ikonen

muratkugic commented on Source/XUnitTest/GameBoardTest.cs:
```diff
@@ -0,0 +1,68 @@
+
+
+    }
+}
```
snyggt!! : ) 👍

muratkugic commented on Source/XUnitTest/GameSessionTest.cs:
```diff
@@ -0,0 +1,61 @@
+            Assert.True(string.IsNullOrWhiteSpace(gameSession.PlayerName));
+        }
+    }
+}
```
Återigen! vackert att se! : )

muratkugic commented on Source/XUnitTest/PlayerTest.cs:
```diff
@@ -0,0 +1,32 @@
+            Assert.True(diceNumber > 0 && diceNumber < 7);
+        }
+    }
+}
```
👍

Adgnascor reviewed (commented):
kommenterat via Murres konto :)

Irvper77 reviewed (commented):
Bra jobbat  bra att ni kommit så långt och snyggt med tester 👍

erifredrika reviewed (commented):
Som sagt tidigare fint jobb med planeringen! Snyggt med tester också. Fortsätt så :)

AntonNorgren commented on Source/LudoGame/Menu.cs:
```diff
@@ -0,0 +1,149 @@
+
+        private static IDictionary<string, int> playerProfiles = new Dictionary<string, int>();
+
+        public static void Display()
```
Hela menyn är snyggt uppbyggd med lättläst kod. Bra jobbat

AntonNorgren commented on Source/LudoGameEngine/AppSettings.Json:
```diff
@@ -0,0 +1,5 @@
+﻿{
+    "ConnectionStrings": {
+        "DefaultConnection": "SERVER=den1.mssql8.gear.host; DATABASE=ludogame; UID= ludogame;PWD=Yw7M8dSu_b_q;"
```
Lägg till GitIgnore på denna filen!

skjohansen replied:
Och byt lösenord efteråt!!

AntonNorgren commented on Source/LudoGameEngine/GameBoard.cs:
```diff
@@ -0,0 +1,292 @@
+        private int DecidePlayerStart(int pAmount)
+        {
+            
+            IDictionary<int, int> playersThrow = new Dictionary<int, int>();
```
Smart lösning med dictionary istället för lista.

AntonNorgren commented on Source/LudoGameEngine/GameBoard.cs:
```diff
@@ -0,0 +1,292 @@
+        }
+
+        //work in progress
+        public IList<GamePlayer> SetPlayOrder(int id, IList<GamePlayer> gp)
```
Behövs detta verkligen göras? Görs inte detta redan i DecideStartingPlayer.

aohzork committed:
Driver@aohzork

AntonNorgren reviewed (commented):
Överlag bra upplägg. Ska bli spännande att se slutresultat. Bra jobbat med tester.

aohzork committed:
Driver @aohzork

skjohansen commented on Source/LudoGame/Menu.cs:
```diff
@@ -0,0 +1,149 @@
+                         //ChoosePlayerColor().
+                         //SetPlayerPositions().
+                         SaveState().
+                         StartGame();
```
Nice solution!!!
I assume that SetPalyerAmount and SetSessionData take a console input, could this be seperated?? To something like:
SetPlayerAmount(ReadPlayerAmountFromConsole()) etc

aohzork committed:
Driver @aohzork

skjohansen commented on Source/XUnitTest/GameBoardTest.cs:
```diff
@@ -0,0 +1,68 @@
+            IGameSession gs = new GameSession();
+            GameBoard gameBoard = new GameBoard(gs);
+
+            if (color == "Red")
+                expectedValue = 1;  //ev. 0 om man räknar från index 0
+            else if (color == "Blue")
+                expectedValue = 11;  //ev. 10 om man räknar från index 0
+            else if (color == "Green")
+                expectedValue = 21;  //ev. 20 om man räknar från index 0
+            else if (color == "Yellow")
+                expectedValue = 31;  //ev. 30 om man räknar från index 0
```
ExpectedValue should be part of the InlineData
```c#
      [Theory]
        [InlineData("Red", 1)]
        [InlineData("Blue", 11)]
        [InlineData("Green", 21)]
        [InlineData("Yellow", 31)]
        public void SetColorStartPositon_InputColor_ReturnCorrectBoardPosition(string color, int expectedValue)
```

skjohansen commented on Source/XUnitTest/GameSessionTest.cs:
```diff
@@ -0,0 +1,61 @@
+                PlayerAmount = input
+            };
+
+            Assert.True(gameSession.PlayerAmount > 1 && gameSession.PlayerAmount < 5);
```
I assume that the you expect what you out in?
I'm a bit unsure why this check (with the > && <) is needed
```c#
        [Theory]
        [InlineData(2)]
        [InlineData(3)]
        [InlineData(4)]
        public void SetPlayerAmount_ValidAmount_True(int input)
        {
            GameSession gameSession = new GameSession
            {
                PlayerAmount = input
            };

            Assert.Equals(input, gameSession.PlayerAmount);
```

skjohansen commented on Source/XUnitTest/PlayerTest.cs:
```diff
@@ -0,0 +1,32 @@
+            Type expected = value.GetType();
+
+            Assert.IsType(expected, diceNumber);
+            return diceNumber;
```
A unit test with a return value?

skjohansen commented on Source/XUnitTest/PlayerTest.cs:
```diff
@@ -0,0 +1,32 @@
+        [Fact]
+        public void DiceRoll_IntValueBetweenOneAndSix()
+        {
+            Random number = new Random();
+
+            int diceNumber = number.Next(1, 7);
+
+            Assert.True(diceNumber > 0 && diceNumber < 7);
```
You don't need to test this, Microsoft has hopefully allready tested the Random class

skjohansen commented on Source/XUnitTest/PlayerTest.cs:
```diff
@@ -0,0 +1,32 @@
+        {
+            Random number = new Random();
+
+            int diceNumber = number.Next(1, 7);
+
+            int value = 0;
+            Type expected = value.GetType();
+
+            Assert.IsType(expected, diceNumber);
```
Do you test if an int is an int?

skjohansen reviewed (commented):
I like the way you are working with unittests, and in general does seem that you have a good idea of what you are building. But the code can many places be hard to understand, especially look at you naming, you nesting of methods and split you code into multiple classes!

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

samirehsani19 committed:
Driver @samirehsani19

Har gjort färdig databas delen.
man kan inserta data
läsa data
ta bort data
ladda spel
hitta bästa spelare
redigera data

aohzork committed:
Driver@aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
Driver @samirehsani19

Solved merge conflict

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

PontusGillenang committed:
Driver @pontusgillenang

Some test cleaning

PontusGillenang committed:
Revert "Driver @pontusgillenang"

This reverts commit 476dd9b5e5281be8f097ea433e5a5068eff4c50f.

skjohansen committed:
Classroom fixes

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

felhanterat och förbjudit samma session namn, och obehöriga vinnare till en session

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
Driver @samirehsani19

checked everything works.

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

lots of refractoring & worked alot on design

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Revert "Driver @pontusgillenang"

This reverts commit 65ae0587300ea49b1af3c1bee0e771b7ed5a4ef8.

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

lots of fixes, refactoring code and implement changes & features

aohzork committed:
Driver @aohzork

minor adjustments

samirehsani19 committed:
Driver @samirehsani19

contected some of database class and methods to menu

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
driver @samirehsani19

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Created method for session names

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@ aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

can move pieces to goal, current global position and new global position

aohzork committed:
Driver @aohzork

different colors on common board move around independently

aohzork committed:
Driver @aohzor

Load Game works from creating game with pieces position 0, from preinserted db-content

aohzork committed:
Driver@aohzork

ingame save and save winner methods prepared,  making some code more readable.

aohzork committed:
Driver @aohzork

implemented winner and activate save game functions

aohzork committed:
Driver @aohzork

small fixes in gamesession

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

PontusGillenang committed:
Driver @pontusgillenang

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Revert "Driver @pontusgillenang"

This reverts commit df37b1bacdd74cfba2dc54ba86f7cd9f3f476284.

aohzork committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

simplfied movementmethods in earlier push, and catalogueizing some  in this.

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

aohzork committed:
Driver @aohzork

knockout other player

aohzork committed:
Driver @aohzork

move behind some more work

aohzork committed:
Driver @aohzork

working move behind & knockout + win

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

samirehsani19 committed:
Driver @samirehsani19

solved dubblicate names and position of piece

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

rättat till koden i loopen

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @ohzork

updating documentation and cleaning som code

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

cleanup

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Snyggat till utskriften i game load

aohzork committed:
Driver @aohzork

refactoring & cleaning up

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Solved a problem with saving the winner and solved other small problems

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

some fixes

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

uppdatering av klassdiagram

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

uppdaterar klassinformation

aohzork committed:
Driver @ aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

solved problem with position updating

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

updating klassbeskrivning & separation of code

samirehsani19 committed:
Driver @samirehsani19

Created wining song

samirehsani19 committed:
Driver @samirehsnai19

aohzork committed:
Driver @aohzork
